rmf_websocket/src/rmf_websocket/BroadcastClient.cpp

```diff
@@ -65,16 +134,23 @@ class BroadcastClient::Implementation
   {
     /// _queue is thread safe. No need to lock.
     _queue.push(msg);
+    _io_service.dispatch([this]()
+      {
+        _flush_queue_if_connected();
+      });
   }
 
   //============================================================================
   void publish(const std::vector<nlohmann::json>& msgs)
   {
     for (auto msg: msgs)
     {
-      /// _queue is thread safe. No need to lock.
       _queue.push(msg);
     }
+    _io_service.dispatch([this]()
```

Comment:
Consider logging something when the buffer overflows, to avoid race conditions, ringbuffer needs to return if an overflow occurred on `push`.

Reply:
Can I clarify what the race condition would be? Wouldnt we just be dropping the oldest messages?

Reply:
The check for if a push cause an overflow needs to an atomic, else it is possible that something like this happens

1. thread1 sees that the buffer is not full
2. thread2 pushes a message into the buffer, causing it to be full
3. thread1 pushes a message, it thinks the buffer is not full so it assumes it does not cause an overflow

In this scenario, we are never aware that a message was dropped.